main.tf

```diff
@@ -4,7 +4,7 @@ locals {
 
 resource "datadog_monitor" "generic_datadog_monitor" {
   count = var.enabled ? 1 : 0
-  name  = "${var.service} - ${var.name}"
+  name  = "${var.name_prefix}${var.service} - ${var.name}${var.name_suffix}"
```

Comment:
What will be the use case for the prefix and suffix?

I was thinking, probably you want to use the same separator that we always use. Then this would be more convenient, but of course that depends on the actual use case:

```hcl
  name  = join(" - ", compact(
    [
      var.name_prefix,
      var.service,
      var.name,
      var.name_suffix
    ]
  ))
```